src/cbrkit/sim/graphs/alignment.py

```diff
@@ -0,0 +1,191 @@
+        ... })
+        >>> graph_y = Graph(nodes=nodes_y, edges=edges_y, value=None)
+        >>> g_dtw = dtw(node_similarity, edge_similarity)
+        >>> result = g_dtw(graph_x, graph_y)
+        >>> print(result)
+        GraphSim(value=0.19444444444444442, node_mappings={'1': '1', '2': '2'}, edge_mappings={'e1': 'e1'})
+    """
+
+    node_sim_func: Callable[[Any, Any], float]
+    edge_sim_func: Callable[[Any, Any], float] | None = None
+    normalize: bool = True
+
+    def __call__(
+        self,
+        x: Graph[K, Any, Any, Any],
+        y: Graph[K, Any, Any, Any],
+    ) -> GraphSim[K]:
+        """
+        Perform Graph-DTW using node and optional edge similarity functions.
+
+        Args:
+            x: The case graph.
+            y: The query graph.
+
+        Returns:
+            A GraphSim object containing the similarity score and mappings.
+        """
+        if not (is_sequential(x) and is_sequential(y)):
+            raise ValueError("Input graphs must be sequential workflows")
+
+        sequence_x = self.get_sequential_nodes(x)
+        sequence_y = self.get_sequential_nodes(y)
+
+        # Use the dtwmodule for node distances
+        node_result = dtwmodule(distance_func=self.node_sim_func)(
+            sequence_x, sequence_y, return_alignment=True
+        )
+
+        # Extract node distance (value) and alignment (local_similarities)
+        node_distance = node_result.value
+        alignment = node_result.local_similarities
+
+        node_similarity = dist2sim(node_distance)
+
+        node_mappings = {
+            y_node.key: x_node.key for x_node, y_node in alignment if x_node and y_node
+        }
+
+        edge_similarity = None
+        edge_mappings = {}
+
+        if self.edge_sim_func:
+            edge_sequence_x = self.get_sequential_edges(x, sequence_x)
+            edge_sequence_y = self.get_sequential_edges(y, sequence_y)
+
+            # Use the dtwmodule for edge distances
+            edge_result = dtwmodule(distance_func=self.edge_sim_func)(
+                edge_sequence_x, edge_sequence_y, return_alignment=False
+            )
+
+            # Extract edge distance (value)
+            edge_distance = edge_result.value
+
+            edge_similarity = dist2sim(edge_distance)
+
+            edge_mappings = {
+                y_edge.key: x_edge.key
+                for x_edge, y_edge in zip(edge_sequence_x, edge_sequence_y)
+                if x_edge and y_edge
+            }
+
+        total_similarity = (
+            (node_similarity + edge_similarity) / 2.0
+            if edge_similarity is not None
+            else node_similarity
+        )
+
+        if self.normalize and len(alignment) > 0:
+            total_similarity /= len(alignment)
+
+        return GraphSim(
+            value=total_similarity,
+            node_mappings=node_mappings,
+            edge_mappings=edge_mappings,
+        )
+
+    def get_sequential_nodes(
```

Comment:
Should we move this function to `cbrkit.sim.graphs.model`? I think it could be useful elsewhere